src/v3/src/transformer/uischema/createIdentifierContainer.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IDX_STEP } from 'src/constants';
+
+import { IdentifierContainerElement, TransformStepFnWithOptions } from '../../types';
+import { traverseLayout } from '../util';
+
+export const createIdentifierContainer: TransformStepFnWithOptions = ({
+  transaction,
+  widgetProps,
+}) => (formbag) => {
+  const transactionIdentifier: string | undefined = transaction
+    ?.context?.user?.value?.identifier as string;
+  const { features } = widgetProps;
+  let hasIdentifierContainer = false;
+
+  const shouldHideIdentifier = (
+    showIdentifier?: boolean,
+    identifier?: string,
+    stepName?: string,
+  ): boolean => {
+    const excludedSteps = [IDX_STEP.IDENTIFY, IDX_STEP.CONSENT_ADMIN];
+    // Should not display identifier here because if invalid identifier
+    // is used, introspect includes the invalid name in user context
+    if (typeof stepName !== 'undefined' && excludedSteps.includes(stepName)) {
+      return true;
+    }
+
+    if (showIdentifier === false) {
+      return true;
+    }
+
+    if (!identifier) {
+      return true;
+    }
+
+    return false;
+  };
+
+  // Traverse existing layout and check for existing IdentifierContainers
+  traverseLayout({
+    layout: formbag.uischema,
+    predicate: (el) => (el.type === 'IdentifierContainer'),
+    callback: (_) => {
+      hasIdentifierContainer = true;
+    },
+  });
+
+  // If we don't find an IdentifierContainer that has been added by a custom transformer, add an
```

Comment:
Can we always add an identifier container only using this transformer instead of needing to check for one added by a custom transformer?